web/error/handler.go

```diff
@@ -0,0 +1,57 @@
+package error
+
+import (
+	"net/http"
+
+	"github.com/labstack/echo/v4"
+	"github.com/rs/zerolog"
+)
+
+func Handler(logger zerolog.Logger) echo.HTTPErrorHandler {
+	ll := logger.With().Str("middleware", "errorHandler").Logger()
+	return func(err error, c echo.Context) {
+		if c.Response().Committed {
+			return
+		}
+
+		ll.Err(err).
+			Str("requestID", c.Request().Header.Get(echo.HeaderXRequestID)).
+			Msg("request returned an error")
```

Comment:
Change 1 is here. Regardless of what the error is, it needs to be logged. The logger on line 54 would log the error that gets returned from `c.JSON` or `c.NoContent`.